### PR TITLE
Edit camera state variables

### PR DESCRIPTION
**Pull request by RubioJr9**
Converted CameraLookAt and CameraRotation to strings so they can be set through the python interface.

### CODE REVIEW

File: src/Interface/Modules/Render/ViewScene.cc [hunk 1]
```diff
@@ -522,10 +522,20 @@ void ViewSceneDialog::pullCameraLookAt()
   auto spire = mSpire.lock();
   if(!spire) return;
 
-  auto lookAt = toDoubleVector(state_->getValue(Modules::Render::ViewScene::CameraLookAt).toVector());
+  auto lookAt = pointFromString(state_->getValue(Modules::Render::ViewScene::CameraLookAt).toString());
   spire->setCameraLookAt(glm::vec3(lookAt[0], lookAt[1], lookAt[2]));
 }
 
+//--------------------------------------------------------------------------------------------------
+glm::quat ViewSceneDialog::stringToQuat(std::string &s)
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.cc]: Move this function to a new header/cc file, as a global function or in a new class, called ViewSceneUtility or the like.

File: src/Interface/Modules/Render/ViewScene.h [hunk 2]
```diff
@@ -39,6 +39,7 @@ DEALINGS IN THE SOFTWARE.
 #include <Interface/Modules/Render/ViewSceneControlsDock.h>
 #include <Graphics/Datatypes/GeometryImpl.h>
 #include <Interface/Modules/Render/share.h>
+#include <glm/gtc/quaternion.hpp>
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.h]: Part of: Move this function to a new header/cc file, as a global function or in a new class, called ViewSceneUtility or the like.

Two problems here: the glm is after the share.h. Also, we want to minimize glm includes in our header code.

File: src/Interface/Modules/Render/ViewScene.h [hunk 3]
```diff
@@ -124,6 +125,7 @@ namespace SCIRun {
       void pullCameraRotation();
       void pullCameraLookAt();
       void pullCameraDistance();
+      glm::quat stringToQuat(std::string &s);
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.h]:
```suggestion
      glm::quat stringToQuat(const std::string &s);
```

Move this function to a new header/cc file, as a global function or in a new class, called ViewSceneUtility or the like.

File: src/Interface/Modules/Render/ViewScene.cc [hunk 4]
```diff
@@ -522,10 +522,20 @@ void ViewSceneDialog::pullCameraLookAt()
...
+{
+  std::istringstream is(s);
+  double w, x, y, z;
+  char st;
+  is >> st >> w >> st >> x >> st >> y >> st >> z >> st;
```
Comment (dcwhite) [inline src/Interface/Modules/Render/ViewScene.cc]: This is also a bit brittle when Python input is considered. Look into `boost::regex` to handle the intervening commas/spaces better. And add a unit test.

**Comment by jessdtate** [comment]
The python behavior seems to be fine, but I can't seem to rotate the camera with the mouse.  Is anyone else seeing this?

**Comment by dcwhite** [comment]
@jessdtate I tested Mac and Windows and had no problem with mouse rotation. Did you move it with python first? What were the exact testing steps?

**Comment by dcwhite** [comment]
@RubioJr9 This also needs better error-checking. I set the rotation parameter to a "Point" (having just tested sending a point to CameraLookAt), and got this bizarrely rotated object (rendering a basic latvol):
image

**Comment by dcwhite** [comment]
Then I set the rotation to a "5-tuple" string, and got this view, which I could not move/zoom away from (might have been what @jessdtate encountered):
image

**Comment by dcwhite** [comment]
I'm going to close this one too and wait for a redo, with the changes mentioned above and better error-checking. It's not ready for milestone X.1.

**Comment by RubioJr9** [comment]
How do I reopen a PR?

**Comment by dcwhite** [comment]
Just make a new one.

**Pull request closed without merging**